instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -88,22 +88,24 @@ defmodule OpentelemetryEcto do
           acc
       end)
 
-    s =
-      OpenTelemetry.Tracer.start_span(span_name, %{
-        start_time: start_time,
-        attributes: Map.merge(attributes, base_attributes),
-        kind: :client
-      })
-
-    case query_result do
-      {:error, error} ->
-        OpenTelemetry.Span.set_status(s, OpenTelemetry.status(:error, format_error(error)))
-
-      {:ok, _} ->
-        :ok
-    end
+    current_ctx = OpenTelemetry.Ctx.get_current()
+
+    parent_ctx =
+      if current_ctx == %{} do
+        OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
+      else
+        current_ctx
+      end
 
-    OpenTelemetry.Span.end_span(s)
+    OpenTelemetry.Tracer.with_span parent_ctx, span_name, %{
+      start_time: start_time,
+      attributes: Map.merge(attributes, base_attributes),
+      kind: :client
+    } do
+      with {:error, error} <- query_result do
```

Comment:
I just realized this isn't doing anything. Why was a span started and ended in this handle event if nothing is happening here? Did you mean to also set the time it ended?

I would not use `with_span` here since you aren't actually wrapping the logic you want to have a span for but I guess are creating a span for something that already happened?

Reply:
Yes, the event is emitted once for every query, so indeed, it's creating a span for something that already happened (hence the `start_time`, which is computed from the `total_time` measurement).